onnxruntime/contrib_ops/cpu/bert/rotary_embedding.cc

```diff
@@ -98,16 +62,16 @@ Status RotaryEmbedding<T>::Compute(OpKernelContext* context) const {
 
       const int block_offset = b * batch_stride + s * seq_stride + n * head_stride;
 
-      const T* input_data = input_src + block_offset;
-      T* output_data = output_dest + block_offset;
+      const T* input_data = input + block_offset;
+      T* output_data = output + block_offset;
 
       // Cache is (M, H/2) or (M, rotary_embedding_dim/2)
       const int position_id = (position_ids_format == 0)
-                                  ? static_cast<int>(pos_ids_data[0]) + s
-                                  : static_cast<int>(pos_ids_data[b * sequence_length + s]);
+                                  ? static_cast<int>(position_ids[0]) + s
+                                  : static_cast<int>(position_ids[b * sequence_length + s]);
```

Comment:
can we calculate the position_id directly?

Reply:
unless we have past_seq_len and some padding information (like whether it is left/right side padding).